tooling/miniutil/src/build/mod.rs

```diff
@@ -73,6 +84,32 @@ impl ProgramBuilder {
         self.functions.try_insert(name, f).unwrap();
         name
     }
+
+    pub fn declare_vtable(&mut self, trait_name: TraitName, ty: Type) -> VTableBuilder {
+        let name = VTableName(Name::from_internal(self.next_vtable));
+        self.next_vtable += 1;
+        VTableBuilder::new(
+            trait_name,
+            name,
+            ty.layout::<DefaultTarget>().expect_size("only sized types can be trait objects"),
+            ty.layout::<DefaultTarget>().expect_align("only sized types can be trait objects"),
+        )
+    }
+
+    #[track_caller]
+    pub fn finish_vtable(&mut self, v: VTableBuilder) -> VTableName {
+        let name = v.name();
+        let vtable = v.finish_vtable();
+        // We could store and check the number of functions matches what was defined here.
```

Comment:
Is this a TODO or FIXME, or what is the point of this comment?

Reply:
In that case I'd phrase it more like
```suggestion
        // Currently, we do not check that this vtable has all the methods required by the trait.
```